bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.magentatv.internal.network;
+
+import static org.openhab.binding.magentatv.internal.MagentaTVBindingConstants.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.text.MessageFormat;
+import java.util.Properties;
+import java.util.UUID;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.magentatv.internal.handler.MagentaTVControl;
+import org.openhab.binding.magentatv.internal.handler.MagentaTVGson.MROAuthGson.OAuthAutenhicateResponse;
+import org.openhab.binding.magentatv.internal.handler.MagentaTVGson.MROAuthGson.OAuthTokenResponse;
+import org.openhab.binding.magentatv.internal.handler.MagentaTVGson.MROAuthGson.OauthCredentials;
+import org.openhab.binding.magentatv.internal.handler.MagentaTVGson.MROAuthGson.OauthKeyValue;
+import org.openhab.binding.magentatv.internal.utils.MagentaTVException;
+import org.openhab.binding.magentatv.internal.utils.MagentaTVLogger;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link MagentaTVOAuth} class implements the OAuth authentication, which
+ * is used to query the userID from the Telekom platform.
+ *
+ * @author Markus Michels - Initial contribution
+ * @author Mathias Gisch - initial bash script
+ *
+ *         Deutsche Telekom uses a OAuth-based authentication to access the EPG portal. The
+ *         communication between the MR and the remote app requires a pairing before the receiver could be
+ *         controlled by sending keys etc. The so called userID is not directly derived from any local parameters
+ *         (like terminalID as a has from the mac address), but will be returned as a result from the OAuth
+ *         authentication. This will be performed in 3 steps
+ *         1. Get OAuth credentials -> Service URL, Scope, Secret, Client ID
+ *         2. Get OAth Token -> authentication token for step 3
+ *         3. Authenticate, which then provides the userID (beside other parameters)
+ *
+ */
+@NonNullByDefault
+public class MagentaTVOAuth {
+    private final MagentaTVLogger logger = new MagentaTVLogger(MagentaTVOAuth.class, "OAuth");
+
+    @SuppressWarnings("null")
+    public String getOAuthCredentials(String accountName, String accountPassword) throws MagentaTVException {
+        logger.info("Authenticate with account {0}", accountName);
```

Comment:
change to logger .debug

Reply:
that's an important info, it shows the user account